rinja_derive/src/integration.rs

```diff
@@ -223,3 +232,265 @@ fn string_escape(dest: &mut String, src: &str) {
+        let Some(mut var_args) = var_args else {
+            continue;
+        };
+
+        let var_ast = type_for_enum_variant(enum_ast, &generics, var);
+        buf.write(quote!(#var_ast).to_string());
+
+        // not inherited: template, meta_docs, block, print
+        if let Some(enum_args) = &mut enum_args {
+            set_default(&mut var_args, enum_args, |v| &mut v.source);
+            set_default(&mut var_args, enum_args, |v| &mut v.escape);
+            set_default(&mut var_args, enum_args, |v| &mut v.ext);
+            set_default(&mut var_args, enum_args, |v| &mut v.syntax);
+            set_default(&mut var_args, enum_args, |v| &mut v.config);
+            set_default(&mut var_args, enum_args, |v| &mut v.whitespace);
+        }
+        let size_hint = biggest_size_hint.max(build_template_item(
+            buf,
+            &var_ast,
+            &TemplateArgs::from_partial(&var_ast, Some(var_args))?,
+            TmplKind::Variant,
+        )?);
+        biggest_size_hint = biggest_size_hint.max(size_hint);
+
+        variant_as_arm(
+            &var_ast,
+            var,
+            size_hint,
+            &mut render_into_arms,
+            &mut size_hint_arms,
+        );
+    }
+    if has_default_impl {
+        let size_hint = build_template_item(
+            buf,
+            enum_ast,
+            &TemplateArgs::from_partial(enum_ast, enum_args)?,
+            TmplKind::Variant,
+        )?;
+        biggest_size_hint = biggest_size_hint.max(size_hint);
+
+        render_into_arms.extend(quote! {
+            ref __rinja_arg => {
+                <_ as rinja::helpers::EnumVariantTemplate>::render_into(__rinja_arg, writer)
+            }
+        });
+        size_hint_arms.extend(quote! {
+            _ => {
+                #size_hint
+            }
+        });
+    }
+
+    write_header(enum_ast, buf, "rinja::Template");
+    buf.write(format_args!(
+        "\
+        fn render_into<RinjaW>(&self, writer: &mut RinjaW) -> rinja::Result<()>\
+        where \
+            RinjaW: rinja::helpers::core::fmt::Write + ?rinja::helpers::core::marker::Sized\
+        {{\
+            match *self {{\
+                {render_into_arms}\
+            }}\
+        }}\
+        \
+        fn render(&self) -> rinja::Result<rinja::helpers::std::string::String> {{\
+            let size_hint = match self {{\
+                {size_hint_arms}\
+            }};\
+            let mut buf = rinja::helpers::std::string::String::new();\
+            let _ = buf.try_reserve(size_hint);\
+            self.render_into(&mut buf)?;\
+            Ok(buf)\
+        }}\
+        \
+        const SIZE_HINT: rinja::helpers::core::primitive::usize = {biggest_size_hint}usize;\
+        \
+        }}\
+        }};",
+    ));
+    Ok(biggest_size_hint)
+}
+
+fn set_default<S, T, A>(dest: &mut S, parent: &mut S, mut access: A)
+where
+    T: Clone,
+    A: FnMut(&mut S) -> &mut Option<T>,
+{
+    if let dest @ None = access(dest) {
```

Comment:
It's super weird. What about:

```suggestion
    if access(dest).is_none() {
```

?

Reply:
It's the same as `let dest = access(dest); if dest.is_none() {` in one line. I found it nifty. :) But changed.